### PR TITLE
Support step filter when stepping

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

Add debugFilters options in launch.json to allow user to customize the filters to debug "Just My Code".

### CODE REVIEW

**Comment by testforstephen** [comment]
Peer PR on server side: https://github.com/Microsoft/java-debug/pull/106

stepfilter

File: README.md [hunk 1]
```diff
@@ -56,6 +56,11 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
 - `projectName` - The preferred project in which the debugger searches for classes. There could be duplicated class names in different projects. This setting also works when the debugger looks for the specified main class when launching a program.
 - `cwd` - The working directory of the program.
 - `env` - The extra environment variables for the program.
+- `debugFilters` - Customize the filters to debug "Just My Code".
```
Comment (yaohaizh) [inline README.md]: Should avoid the "Just My Code". The "Just My Code" is .net term and has different behavior from here. We should avoid the confusion.

File: README.md [hunk 2]
```diff
@@ -56,6 +56,11 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
...
+  - `stepFilters` - Skip these classes when stepping. Each filter could be fully qualified class name or pattern. Example: [\"java.lang.ClassLoader\", \"java.*\"].
```
Comment (yaohaizh) [inline README.md]: Should be more detail information about the pattern that can be applied here? Regex?

File: README.md [hunk 3]
```diff
@@ -56,6 +56,11 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
...
+  - `skipSynthetics` - Skip synthetic methods when stepping.
```
Comment (yaohaizh) [inline README.md]: What is synthetic methods?

Comment (testforstephen) [reply README.md]: https://javapapers.com/core-java/java-synthetic-class-method-field/
https://www.javaworld.com/article/2073578/java-s-synthetic-methods.html

File: README.md [hunk 4]
```diff
@@ -61,6 +61,11 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
   - `internalConsole` - VS Code debug console (input stream not supported).
   - `integratedTerminal` - VS Code integrated terminal.
   - `externalTerminal` - External terminal that can be configured in user settings.
+- `stepFilters` - Customize the filters to skip the specified methods and classes when stepping.
```
Comment (akaroml) [inline README.md]: Indention

Comment (akaroml) [reply README.md]: "Skip specified classes or methods when stepping."

File: README.md [hunk 5]
```diff
@@ -61,6 +61,11 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
...
+  - `classNameFilters` - Skip these classes when stepping. Each filter could be fully qualified class name or regular expression that begins with '\*' or ends with '\*'. Example: [\"java.lang.ClassLoader\", \"java.*\"].
```
Comment (akaroml) [inline README.md]: Skip the specified classes when stepping. Class names should be fully qualified. Wildcard is supported.

**Pull request merged**